File: dxa.java/ecommerce-framework-dxa-module/src/main/java/com/sdl/ecommerce/dxa/model/ECommerceFilterAttribute.java
package com.sdl.ecommerce.dxa.model;

import com.sdl.ecommerce.api.QueryFilterAttribute;
import com.sdl.webapp.common.api.mapping.semantic.annotations.SemanticEntity;
import com.sdl.webapp.common.api.mapping.semantic.annotations.SemanticProperty;
import com.sdl.webapp.common.api.model.entity.AbstractEntityModel;

import static com.sdl.webapp.common.api.mapping.semantic.config.SemanticVocabulary.SDL_CORE;

/**
 * ECommerce Filter Attribute
 *
 * @author nic
 */
@SemanticEntity(entityName = "ECommerceFilterAttribute", vocabulary = SDL_CORE, prefix = "e", public_ = false)
public class ECommerceFilterAttribute extends AbstractEntityModel {

    @SemanticProperty("e:name")
    private String name;

    @SemanticProperty("e:value")
    private String value;

    @SemanticProperty("e:mode")
    private String mode;

    public String getName() {
        return name;
    }

    public String getValue() {
        return value;
    }

    public String getMode() {
        return mode;
    }

    /**
     * Convert to filter attribute to be used in E-Commerce queries.
     * @return filter attribute
     */
    public QueryFilterAttribute toQueryFilterAttribute() {
        QueryFilterAttribute.FilterMode filterMode;
        if ( mode != null && mode.equals("exclude") ) {
            filterMode = QueryFilterAttribute.FilterMode.EXCLUDE;
        }
        else {
            filterMode = QueryFilterAttribute.FilterMode.INCLUDE;
        }
        return new QueryFilterAttribute(name, value, filterMode);
    }
}

File: dxa.net/ecommerce-framework-dotnet/SDL.ECommerce.DXA/Models/ECommerceFilterAttribute.cs
using Sdl.Web.Common.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace SDL.ECommerce.DXA.Models
{
    [SemanticEntity(Vocab = CoreVocabulary, EntityName = "ECommerceFilterAttribute", Prefix = "e")]
    public class ECommerceFilterAttribute : EntityModel
    {
        [SemanticProperty("e:name")]
        public string Name { get; set; }

        [SemanticProperty("e:value")]
        public string Value { get; set; }

        [SemanticProperty("e:mode")]
        public string Mode { get; set; }
    }
}